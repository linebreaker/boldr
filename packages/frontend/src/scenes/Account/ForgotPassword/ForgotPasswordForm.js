/* @flow */

import React from 'react';
import { Field, reduxForm } from 'redux-form';
import Button from '@boldr/ui/Button';
import Form, { TextFormField } from '@boldr/ui/Form';

const style = {
  margin: 12,
};
export type Props = {
  handleSubmit: Function,
  submitting?: boolean,
  pristine?: boolean,
};
const ForgotPasswordForm = (props: Props) => {
  const { handleSubmit, pristine, submitting } = props;
  return (
    <Form onSubmit={handleSubmit} className="boldrui-form boldr-form--forgot">
      <Field
        id="email"
        name="email"
        type="email"
        placeholder="admin@boldr.io"
        label="Email Address"
        component={TextFormField}
      />
      <Button
        htmlType="submit"
        style={style}
        kind="primary"
        disabled={submitting || pristine}
        outline
        block
      >
        Send Reset Link
      </Button>
    </Form>
  );
};

export default reduxForm({
  form: 'forgotPasswordForm',
})(ForgotPasswordForm);
