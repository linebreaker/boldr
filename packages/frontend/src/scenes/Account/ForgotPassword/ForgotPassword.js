/* @flow */
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';

import FormCard from '@boldr/ui/Form/FormCard';
import { forgotPassword } from '../state/actions';
import ForgotPasswordForm from './ForgotPasswordForm';

type Props = {
  dispatch: Function,
};

class ForgotPassword extends PureComponent {
  props: Props;

  handleSubmit = values => {
    const { email } = values;
    this.props.dispatch(forgotPassword(email));
  };
  render() {
    return (
      <div>
        <Helmet title="Forgot Password" />
        <div className="forgotpw-wrapper">
          <FormCard
            title="Forgot Password"
            skinny
            lightText
            form={<ForgotPasswordForm onSubmit={this.handleSubmit} />}
          />
        </div>
      </div>
    );
  }
}

export default connect()(ForgotPassword);
