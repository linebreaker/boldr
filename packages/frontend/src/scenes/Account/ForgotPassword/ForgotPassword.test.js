import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ForgotPassword from './ForgotPassword';

test('<ForgotPassword />, renders the forgot password form card', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);
  const wrapper = shallow(
    <Provider store={store}>
      <ForgotPassword />
    </Provider>,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
