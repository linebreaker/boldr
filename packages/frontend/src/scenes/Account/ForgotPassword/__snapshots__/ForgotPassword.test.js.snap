// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword />, renders the forgot password form card 1`] = `<Connect(ForgotPassword) />`;
